### ISSUE TITLE
Stride Animation Info

### ISSUE THREAD

**Issue by lxw404** [Yellow-Dog-Man/Resonite-Issues#2872]
### Is your feature request related to a problem? Please describe.

Currently there's seemingly no good way to get the predicted stride position/normals and exactly when a step occurs for each foot from the new animation system.

### Describe the solution you'd like

There should be a way to get:
* Current target stride per foot
* Events per foot that correspond to a "step"

### Describe alternatives you've considered

There are ways of parenting colliders to your feet and coming up with ways to reverse engineer the stride targets, but that's less desirable and not sustainable.

### Additional Context

This would enable cool particle effects, and other such visual systems for when you take steps.

### Requesters

LuxKitty

**Comment by shiftyscales** [comment]
> This would enable cool particle effects, and other such visual systems for when you take steps.

So is the underlying request here just wanting some mechanism to detect where / when an avatar has taken a footstep, @lxw404?

**Comment by lxw404** [comment]
Yes but also being able to know where the system thinks the next footstep will take place.  As shown in the [devlog](https://youtu.be/zjTX7CtXkLo), there are already debug visuals which show where each next stride will happen before you actually take a step which is useful in the case of wanting to place objects/particles there beforehand.

**Comment by Frooxius** [comment]
I plan on adding mechanism to give you footstep events, but not for the stride information - that information doesn't necessarily exist. 

Once a footstep happens, that sends a distinct event which is easy to provide and my goal was to build a system around that once this stuff is settled.
